lib/ansible/plugins/filter/core.py

```diff
@@ -247,20 +247,30 @@ def regex_escape(string, re_type='python'):
 
 
 def from_yaml(data):
+    if data is None:
+        return None
+
     if isinstance(data, string_types):
         # The ``text_type`` call here strips any custom
         # string wrapper class, so that CSafeLoader can
         # read the data
         return yaml_load(text_type(to_text(data, errors='surrogate_or_strict')))
+
+    display.deprecated(f"The from_yaml filter ignored non-string input of type {native_type_name(data)!r}.", version='2.23', obj=data)
     return data
 
 
 def from_yaml_all(data):
+    if data is None:
+        return []  # backward compatibility; ensure consistent result between classic/native Jinja for None/empty string input
```

Comment:
do we want to deprecate the `None` handling also?

Reply:
Since it's existing behavior, I don't see any reason to change this. We're just equating `None` with `""`, which seems correct enough to not cause issues.